util/miner_utils.go

```diff
@@ -0,0 +1,107 @@
+package util
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net"
+	"net/netip"
+	"sync"
+	"time"
+
+	"github.com/INFURA/go-ethlibs/jsonrpc"
+
+	"github.com/dominant-strategies/go-quai-stratum/rpc"
+	"github.com/dominant-strategies/go-quai/core/types"
+)
+
+type MinerSession struct {
+	proto string
+	ip    netip.Addr
+	port  int
+	conn  *net.TCPConn
+	enc   *json.Encoder
+
+	// Stratum
+	sync.Mutex
+	latestId uint64
+}
+
+const (
+	c_Max_Req_Size = 4096
+)
+
+func NewMinerConn(endpoint string) (*MinerSession, error) {
+	localaddr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
```

Comment:
can we parametrize this IP here?

Reply:
Apparently you don't need to create a localaddr when you just want localhost and a random port. Made the change and deleted these lines.